README.md

```diff
@@ -6,27 +6,106 @@
+>   for providing feedback and to ask any question. We appreciate your help
+
+## Usage
+
 The solution provides a web user interface, and a REST API to allow you to
 integrate it in your own applications.
 
+![HomePage Screenshot](docs/images/home-screenshot.png)
+
+See the [user guide](docs/USER_GUIDE.md) to learn how to use the solution and
+the [API Specification](docs/api/README.md) to integrate the solution to your
+own applications.
+
+## Architecture
+
+The goal of the solution is to provide a secure, reliable, performant and cost
+effective tool for finding and removing individual records within objects stored
+in S3 buckets. In order to achieve these goals the solution has adopted the
+following design principles:
+
+1. **Security first:** Every component implements least privilege access,
+   encryption is performed at all layers at rest and in transit, authentication
+   is provided out of the box, expiration of logs is easily configurable, and
+   private data is automatically obfuscated or irreversibly deleted as soon as
+   possible when persisting state.
+2. **Built to scale:** The system has been designed and battle tested for
+   performance and high availability with Data Lakes of TB/PB size.
+3. **Cost optimised:**
+   - **Perform work in batches:** Since the time complexity of removing a single
+     vs multiple records in a single object is practically equal and it is
+     common for data owners to have the legal requirement of removing data
+     within a given _timeframe_, the solution is designed to allow the customer
+     to "queue" multiple matches to be removed in a single job.
+   - **Fail fast:** A deletion job takes place in two distinct phases: Find and
+     Forget. The Find phase queries the objects in your S3 data lakes to find
+     any objects which contain records where a specified column contains at
+     least one of record identifiers (known as **Match IDs**) in the deletion
+     queue. If any queries fail, the job will abandon as soon as possible and
+     the Forget phase will not take place. The Forget Phase takes the list of
+     objects returned from the Find phase, and deletes the relevant rows in only
+     those objects.
+   - **Find, then Forget:** The process of separating the two phases, as
+     opposite of processing the read/write on an object basis, is another cost
+     optimisation technique based on the fact columnar dense formats such as
+     Parquet allow to find matches with increased performance and cost
+     optimisation by reading only relevant columns instead of full content from
+     all columns. By processing only the relevant columns during the Find and
+     deletegating it to a serverless managed service such as Amazon Athena, we
+     can then process full rows only during the Forget phase, restricting the
+     most expensive operation to only the relevant objects.
+   - **Serverless:** Where possible, the solution only uses Serverless
```

Comment:
we should probably state why Serverless optmizes costs here

Reply:
Done